signals.go

```diff
@@ -28,12 +28,16 @@ func newSignalHandler(enableSubreaper bool, notifySocket *notifySocket) *signalH
 	// ensure that we have a large buffer size so that we do not miss any signals
 	// in case we are not processing them fast enough.
 	s := make(chan os.Signal, signalBufferSize)
-	// handle all signals for the process.
-	signal.Notify(s)
-	return &signalHandler{
-		signals:      s,
-		notifySocket: notifySocket,
-	}
+	handler := make(chan *signalHandler)
+	go func() {
```

Comment:
It would be nice to have a comment explaining that `signal.Notify` is actually quite expensive and so we defer creating it to a goroutine for that reason. Something like

```suggestion
	// signal.Notify is actually quite expensive, as it has to configure the
	// signal mask and add signal handlers for all signals (all ~65 of them).
	// So, defer this to a background thread while doing the rest of the io/tty
	// setup.
	go func() {
```

Reply:
Yes, have already added now.